### PR TITLE
fix: 404 booking pages when org support is enabled locally

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
Fix 404 booking pages when org support is enabled locally

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

-  Add localhost:3000 to ALLOWED_HOSTNAMES which is the default recommendation for enabling org support locally.
Screenshot 2023-07-10 at 11 04 35 AM
- Simply restart the server and visit http://localhost:3000/pro. It will give 404. After this change it would work

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/test/lib/next-config.test.ts [hunk 1]
```diff
@@ -186,6 +187,10 @@ describe("next.config.js - Org Rewrite", () => {
       expect(
         orgHostRegExp(subdomainRegExp).exec("org.cal.com")?.groups?.orgSlug
       ).toEqual("org");
+
+      expect(
```
Comment (hariombalhara) [inline apps/web/test/lib/next-config.test.ts:191]: This is just an additional test. Nothing is fixed in Next.config.js

File: packages/features/test/orgDomains.test.ts [hunk 2]
```diff
@@ -30,6 +30,14 @@ describe("Org Domains Utils", () => {
         isValidOrgDomain: false,
       });
     });
+
+    it("should return a non valid org domain for localhost", () => {
```
Comment (hariombalhara) [inline packages/features/test/orgDomains.test.ts:34]: Test for the change. It was failing in main.

**Review by leog — APPROVED** [review]
LGTM, thanks for fixing this

**Review by zomars — APPROVED** [review]
Nice Dx improvements @hariombalhara Nice work!

**Comment by zomars** [comment]
@zomars merged this pull request with [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/10029).

**Pull request merged**